docker/Dockerfile.leap.42.3

```diff
@@ -0,0 +1,59 @@
+#
+# Copyright 2019, Intel Corporation
+#
+# 'recipe' for Docker to build an RPM
+#
+
+# Pull base image
+FROM opensuse/leap:42.3
+MAINTAINER daos-stack <daos@daos.groups.io>
+
+# use same UID as host and default value of 1000 if not specified
+ARG UID=1000
+
+# Add build user (to keep rpmbuild happy)
+ENV USER build
+RUN useradd -u $UID -ms /bin/bash $USER
+RUN groupadd -g $UID $USER
+
+# Install basic tools
+RUN zypper --non-interactive update
+
+# basic building
+ENV OPENSUSE_REPO="https://download.opensuse.org/repositories"
+ENV TOOLS_42_3A="openSUSE:/Tools/openSUSE_42.3/openSUSE:Tools.repo"
+ENV TOOLS_42_3="${OPENSUSE_REPO}/${TOOLS_42_3A}"
+ENV HPC_42_3A="science:/HPC/openSUSE_Leap_42.3/science:HPC.repo"
+ENV HPC_42_3="${OPENSUSE_REPO}/${HPC_42_3A}"
+
+RUN zypper --non-interactive ar -f ${TOOLS_42_3}; \
+    zypper --non-interactive ar -f ${HPC_42_3}; \
```

Comment:
Why is this repo needed at the container level?  Repos for building projects should only be needed inside the chroot using a `--repo ...` command to `build`.

Reply:
It should work to put the repo in the container file and not have to specify it anywhere else.
The HPC_42_3 repository is the first case I found that where that does not work that can be reproduced outside of our build environment so a bug can be filed with 'suse' build.

Nexus integration will help, but needs to be done in a way that allows others to use these files with out having to also specifically have a Nexus server setup exactly the same as ours.

Reply:
> It should work to put the repo in the container file and not have to specify it anywhere else.
> The HPC_42_3 repository is the first case I found that where that does not work that can be reproduced outside of our build environment so a bug can be filed with 'suse' build.

So why is it being added at the container level here if it doesn't work?

The real problem with it at the container level is that the docker image constantly needs refreshing (cachebusting), like on every single build to ensure the repo data is not stale.  Whereas if it's added by build it's only local to the chroot for the build and is added with fresh data on every build.

> Nexus integration will help, but needs to be done in a way that allows others to use these files with out having to also specifically have a Nexus server setup exactly the same as ours.

Yes, agreed.  We absolutely should not be doing things that would require others to set up a Nexus server.